Source/Document Structure/SvgLink.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Text;
+
+namespace Svg
+{
+    [NonSvgElementAttribute("link")]
+    public class SvgLink : NonSvgElement
+    {
+        public SvgLink()
+          : base("link")
+        {
+
+        }
+        public enum RelativeValue
+        {
+            Unknown,
+            Alternate,
+            Author,
+            Help,
+            Icon,
+            License,
+            Next,
+            Pingback,
+            Preconnect,
+            Prefetch,
+            Preload,
+            Prerender,
+            Prev,
+            Search,
+            Stylesheet
+        }
+
+        public override SvgElement DeepCopy()
+        {
+            return DeepCopy<SvgLink>();
+        }
+
+
+        [SvgAttribute("href")]
+        public string Href
+        {
+            get { return GetAttribute<string>("href", false, string.Empty); }
+            set { Attributes["href"] = value; }
+        }
+
+        [SvgAttribute("rel")]
+        public RelativeValue Rel
+        {
+            get { return GetAttribute<RelativeValue>("rel", false, RelativeValue.Unknown); }
+            set { Attributes["rel"] = value; }
+        }
+
+
+
+        public string GetLinkContentAsText(RelativeValue rel)
+        {
+            var stream = GetLinkContentAsStream(rel);
+            if (stream == null)
+                return null;
+
+            string content = null;
+            using (StreamReader sr = new StreamReader(stream))
+            {
+                content = sr.ReadToEnd();
+            }
+            stream.Dispose();
+            return content;
+        }
+
+
+        public Stream GetLinkContentAsStream(RelativeValue rel = RelativeValue.Unknown)
+        {
+            if (Rel != rel && rel != RelativeValue.Unknown)
```

Comment:
If you're doing a comparison with `Rel` property, why not use the value of `Rel` property value ??

Reply:
it is to avoid as much as possible the use of the content with a incorrect rel type.

if you call GetLinkContentAsStream(stylesheet) on a link  with rel="icon" for exemple, you will not get the content and cannot use it as a stylesheet by mistake (and make the code crash).

assuming you are absolutely sure of what you do and what you have, you can just call GetLinkContentAsStream(), you will get the content independently of the rel 

 these technique avoid to check everywhere if the Rel is correct every where you use the link content, here it is done in only one place.

I can remove if if you prefer.